test/unit/specs/markdown-basic.spec.js

```diff
@@ -0,0 +1,104 @@
+import ContentState from '../../../src/muya/lib/contentState'
+import EventCenter from '../../../src/muya/lib/eventHandler/event'
+import ExportMarkdown from '../../../src/muya/lib/utils/exportMarkdown'
+import { MUYA_DEFAULT_OPTION } from '../../../src/muya/lib/config'
+import * as templates from '../markdown'
+
+const defaultOptions = { 'endOfLine': 'lf' }
+const defaultOptionsCrlf = Object.assign({}, defaultOptions, { 'endOfLine': 'crlf' })
+
+const createMuyaContext = options => {
+  const ctx = {}
+  ctx.options = Object.assign({}, MUYA_DEFAULT_OPTION, options)
+  ctx.eventCenter = new EventCenter()
+  ctx.contentState = new ContentState(ctx, ctx.options)
+  return ctx
+}
+
+// ----------------------------------------------------------------------------
+// Muya parser (Markdown to HTML to Markdown)
+//
+
+const verifyMarkdown = (markdown, options) => {
+  const ctx = createMuyaContext(options)
+  ctx.contentState.importMarkdown(markdown)
+
+  const blocks = ctx.contentState.getBlocks()
+  const exportedMarkdown = new ExportMarkdown(blocks).generate()
+
+  // FIXME: We always need to add a new line at the end of the document. Add a option to disable the new line.
```

Comment:
Do you mean we need add an option to disable add new line when generating the markdown text? or we don't need to add new line?

Reply:
I think it's a cool feature to insert a new line at the end of a document but it should exists an option to disable it. I had troubles with pure strings like `const s = '# heading'` initially because I had to insert a new line to each test.